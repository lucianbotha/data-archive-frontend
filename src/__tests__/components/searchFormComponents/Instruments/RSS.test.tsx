import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import RSS from "../../../../components/searchFormComponents/instruments/RSS";

const onChange = jest.fn();
describe("RSS ", () => {
  it("should be defined", () => {
    expect(
      mount(<RSS rss={{ errors: {}, name: "RSS" }} onChange={onChange} />)
    ).toBeDefined();
  });

  it("should render correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(
        mount(<RSS rss={{ errors: {}, name: "RSS" }} onChange={onChange} />)
      )
    ).toMatchSnapshot();
  });
});
