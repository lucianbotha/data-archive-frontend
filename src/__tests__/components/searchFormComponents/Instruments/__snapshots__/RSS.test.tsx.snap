// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rss  should render correctly 1`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  padding: 10px;
}

.c1.c1 {
  text-align: left;
  padding-left: 10px;
}

<RSS
  onChange={[MockFunction]}
  rss={
    Object {
      "errors": Object {},
      "name": "RSS",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "main-grid",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    display: grid;
    grid-template-columns: 50% 50%;
    padding: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="main-grid c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Mode
              </p>
              <SelectField
                name="mode"
                onChange={[Function]}
              >
                <div
                  className="control select "
                >
                  <select
                    name="mode"
                    onChange={[Function]}
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="Fabry Perot"
                      value="Fabry Perot"
                    >
                      Fabry Perot
                    </option>
                    <option
                      key="FP polarimetry"
                      value="FP polarimetry"
                    >
                      FP polarimetry
                    </option>
                    <option
                      key="Imaging"
                      value="Imaging"
                    >
                      Imaging
                    </option>
                    <option
                      key="Polarimetric imaging"
                      value="Polarimetric imaging"
                    >
                      Polarimetric imaging
                    </option>
                    <option
                      key="MOS"
                      value="MOS"
                    >
                      MOS
                    </option>
                    <option
                      key="MOS polarimetry"
                      value="MOS polarimetry"
                    >
                      MOS polarimetry
                    </option>
                    <option
                      key="Spectropolarimetry"
                      value="Spectropolarimetry"
                    >
                      Spectropolarimetry
                    </option>
                    <option
                      key="Spectroscopy"
                      value="Spectroscopy"
                    >
                      Spectroscopy
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Detector Mode
              </p>
              <SelectField
                name="detectorMode"
                onChange={[Function]}
              >
                <div
                  className="control select "
                >
                  <select
                    name="detectorMode"
                    onChange={[Function]}
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="Normal"
                      value="Normal"
                    >
                      Normal
                    </option>
                    <option
                      key="Frame Transfer"
                      value="Frame Transfer"
                    >
                      Frame Transfer
                    </option>
                    <option
                      key="Slot Mode"
                      value="Slot Mode"
                    >
                      Slot Mode
                    </option>
                    <option
                      key="Shuffle"
                      value="Shuffle"
                    >
                      Shuffle
                    </option>
                    <option
                      key="Drift Scan"
                      value="Drift Scan"
                    >
                      Drift Scan
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</RSS>
`;
